ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1109,10 +1110,21 @@ public CompletableFuture<RaftClientReply> setConfigurationAsync(SetConfiguration
         return pending.getFuture();
       }
 
-      getRaftServer().addRaftPeers(peersInNewConf);
-      // add staging state into the leaderState
-      pending = leaderState.startSetConfiguration(request);
+      if(request.getMode() == SetConfigurationRequest.Mode.ADD) {
+         List<RaftPeer> peers = Stream.of(peersInNewConf, new ArrayList<>(current.getAllPeers()))
+            .flatMap(List :: stream).distinct().collect(Collectors.toList());
```

Comment:
What should happen in ADD mode when a new peer already exists in the current peer list?  Should the request fail?

Reply:
I think we only check peers that are not in the current list, other semantics do not change, what do you think?

Reply:
When ADD an existing peer with a different priority, we may either
1. return failure and the configuration remains unchanged, or
2. return success and the configuration changes to the new server priority.

If we do the following, then the user request (priority change) is ignored silently.  It sounds like a bug.
- return success and the configuration remains unchanged.

Reply:
The current code already implements the second point:

1. return success and the configuration changes to the new server priority.

Please take a look at RaftConfigurationImpl#hasNoChange